numba/cuda/tests/cudapy/test_operator.py

```diff
@@ -36,6 +100,137 @@ def test_truediv(self):
+
+                kernel[1, 1](got, arg1[0], arg2[0])
+                expected = op(arg1, arg2)
+                self.assertEqual(got[0], expected)
+
+    @skip_unless_cc_53
+    def test_mixed_fp16_comparison(self):
+        functions = (simple_fp16_gt, simple_fp16_ge,
+                     simple_fp16_lt, simple_fp16_le,
+                     simple_fp16_eq, simple_fp16_ne)
+        ops = (operator.gt, operator.ge, operator.lt, operator.le,
+               operator.eq, operator.ne)
+        types = (np.int8, np.int16, np.int32, np.int64,
+                 np.float32, np.float64)
+        for fn, op, ty in zip(functions, ops, types):
+            with self.subTest(op=op):
+                kernel = cuda.jit(fn)
+
+                expected = np.zeros(1, dtype=np.bool8)
+                got = np.zeros(1, dtype=np.bool8)
+                arg1 = np.random.random(1).astype(np.float16)
+                arg2 = (np.random.random(1) * 100).astype(ty)
+
+                kernel[1, 1](got, arg1[0], arg2[0])
+                expected = op(arg1, arg2)
+                self.assertEqual(got[0], expected)
+
+    @skip_unless_cc_53
+    def test_multiple_float16_comparisons(self):
+        functions = (test_multiple_hcmp_1,
+                     test_multiple_hcmp_2,
+                     test_multiple_hcmp_3,
+                     test_multiple_hcmp_4,
+                     test_multiple_hcmp_5)
+        for fn in functions:
+            with self.subTest(fn=fn):
+                compiled = cuda.jit("void(b1[:], f2, f2, f2)")(fn)
+                ary = np.zeros(1, dtype=np.bool8)
+                arg1 = np.float16(2.)
+                arg2 = np.float16(3.)
+                arg3 = np.float16(4.)
+                compiled[1, 1](ary, arg1, arg2, arg3)
+                self.assertTrue(ary[0])
+
+    @skip_on_cudasim('Compilation unsupported in the simulator')
+    def test_fp16_comparison_ptx(self):
+        functions = (simple_fp16_gt, simple_fp16_ge,
+                     simple_fp16_lt, simple_fp16_le,
+                     simple_fp16_eq, simple_fp16_ne)
+        ops = (operator.gt, operator.ge, operator.lt, operator.le,
+               operator.eq, operator.ne)
+        opstring = ('setp.gt.f16', 'setp.ge.f16',
+                    'setp.lt.f16', 'setp.le.f16',
+                    'setp.eq.f16', 'setp.ne.f16')
+        args = (b1[:], f2, f2)
+
+        for fn, op, s in zip(functions, ops, opstring):
+            with self.subTest(op=op):
+                ptx, _ = compile_ptx(fn, args, cc=(5, 3))
+                self.assertIn(s, ptx)
+
+    @skip_on_cudasim('Compilation unsupported in the simulator')
+    def test_mixed_fp16_comparison_ptx(self):
+        functions = (simple_fp16_gt, simple_fp16_ge,
+                     simple_fp16_lt, simple_fp16_le,
+                     simple_fp16_eq, simple_fp16_ne)
+        ops = (operator.gt, operator.ge, operator.lt, operator.le,
+               operator.eq, operator.ne)
+        types_convert = (np.int8,)
```

Comment:
Is this missing some types?

Reply:
Why did you change the name of the test to `test_fp16_int8_comparison_ptx`? Is there only value in testing with `int8`? I'm trying to understand the reasoning behind testing only this particular type.

If it does only make sense to test with `int8` here, then I don't think you need to iterate over the product of the functions/ops and `types_convert` at all.

Reply:
@gmarkall this test is only comparing against int8 to match the "safe" conversion part of this if condition:

https://github.com/testhound/numba/blob/0b6d045a6c814fa046bb5a5f0c6bb2923712d9e5/numba/cuda/cudadecl.py#L407

I will modify the code to remove the iteration.

Reply:
Code update to remove iteration and comment clarification.